File: javascript01/step05/test06.js
"use strict"

function createScore(name, kor, eng, math) {
  return {
    name: name, //변수를 놓는게 아닌 변수안의 값이 놓여진다
    kor: kor,
    eng: eng,
    sum: kor + eng + math,
    aver: (kor + eng + math) / 3
  }
}
var scores2 = [
  createScore("홍길동1", 100, 100, 100), //함수를 저장하는게 아닌 함수를 호출한후 리턴값을 저장한다
  createScore("홍길동2", 90, 90, 90),
  createScore("홍길동3", 80, 80, 80),
  createScore("홍길동4", 70, 70, 70),
  createScore("홍길동5", 60, 60, 60)
]
console.log("홍길동3의 총점은?", scores2[2].sum)
console.log(scores2)

File: java01/src/step05/Test06.java
package step05;

public class Test06 {
  
  static class Student {
    String name;
    int kor;
    int eng;
    int math;
    int sum;
    float aver;
  }
  public static Student getStudent(int studentNo) {
    
    Student s = new Student();
    s.name = "홍길동";
    s.kor = 100;
    s.eng = 90;
    s.math = 80; 
    s.sum = 100 + 90 + 80;
    s.aver = (100 + 90 + 80) / 3f;
    return s;
  }
  
  public static void main(String[] args) {
    Student s2 = getStudent(100);
    
    System.out.println(s2.name);
    System.out.println(s2.kor);
    System.out.println(s2.eng);
    System.out.println(s2.math);
    System.out.println(s2.sum);
    System.out.println(s2.aver);
  }
}
